[FLINK-29982] move the existing Cassandra connector to external repository

echauchot opened the pull request:
The migration process states that we must have 1 release of Flink with the connector being released both separately and as part of Flink. So this PR is the copy of the existing Cassandra connector (not implementing the new Source API) from the Flink repo to new dedicated Cassandra repo.

The backward compatibility toward Cassandra versions is handled by the cassandra driver so there should be no per-version modules only a single cassandra connector module.

R: @zentol

zentol commented on src/test/java/org/apache/flink/streaming/connectors/cassandra/CassandraConnectorITCase.java:
```diff
@@ -0,0 +1,912 @@
+    // ------------------------------------------------------------------------
+
+    public static CassandraContainer createCassandraContainer() {
+        CassandraContainer cassandra = new CassandraContainer(DockerImageVersions.CASSANDRA_4_0);
```
Don't rely on the Flink DockerImageVersions; this entry will be removed eventually.

echauchot replied:
> We'd like to preserve the git history; can you integrate the `cas_rewrite` branch from this repo into your PR?

That makes total sense. I'll rebase on the branch. IMHO it would be interesting to add an item to your migration doc to inform contributors to keep history, that way you would not need to create the branch yourself.

echauchot replied:
:+1:

zentol commented:
We'd like to preserve the git history; can you rebase your PR against the `cas_rewrite` branch in this repo?

zentol commented on pom.xml:
```diff
@@ -0,0 +1,432 @@
+
+	<groupId>org.apache.flink</groupId>
+	<artifactId>flink-connector-cassandra_${scala.binary.version}</artifactId>
+	<version>4.0-SNAPSHOT</version>
```
```suggestion
	<version>3.0-SNAPSHOT</version>
```

echauchot replied:
:+1:

zentol reviewed (commented):
To make our live easier going forward it would be ideal if the connector would be in it's own directory, similar to the ES connector.
The parent pom doesn't have to contain anything for now.

zentol commented:
> We'd like to preserve the git history; can you integrate the cas_rewrite branch from this repo into your PR?

hold on I may want to update that branch to also contain the docs...

echauchot commented:
Thanks for the very prompt review @zentol !

echauchot commented:
> We'd like to preserve the git history; can you integrate the `cas_rewrite` branch from this repo into your PR?

That makes total sense. I'll rebase on the branch. IMHO it would be interesting to add an item to your migration doc to inform contributors to keep history, that way you would not need to create the branch yourself.

zentol commented:
>  IMHO it would be interesting to add an item to your migration doc to inform contributors to keep history

I've already added this as an item to https://cwiki.apache.org/confluence/display/FLINK/Externalized+Connector+development.

echauchot commented:
> > > We'd like to preserve the git history; can you integrate the `cas_rewrite` branch from this repo into your PR?
> > 
> > 
> > That makes total sense. I'll rebase on the branch. IMHO it would be interesting to add an item to your migration doc to inform contributors to keep history, that way you would not need to create the branch yourself.
> 
> @zentol: I rebased on cas_rerewrite and when I pushed to my fork it has closed the PR automatically with no action from my side. I don't have the rights to re-open it. Could you reopen this PR so that we don't lose your review comments?
> 
> I messed up the rebase. I needed to rebase the other way around

I've fixed my rebase, do you prefer that you reopen this PR and I force push here or that I open another PR ?

echauchot commented:
> > > > We'd like to preserve the git history; can you integrate the `cas_rewrite` branch from this repo into your PR?
> > > 
> > > 
> > > That makes total sense. I'll rebase on the branch. IMHO it would be interesting to add an item to your migration doc to inform contributors to keep history, that way you would not need to create the branch yourself.
> > 
> > 
> > @zentol: I rebased on cas_rerewrite and when I pushed to my fork it has closed the PR automatically with no action from my side. I don't have the rights to re-open it. Could you reopen this PR so that we don't lose your review comments?
> > I messed up the rebase. I needed to rebase the other way around
> 
> @zentol I've fixed my rebase, do you prefer that you reopen this PR and I force push here or that I open another PR ?

I opened a new one: https://github.com/apache/flink-connector-cassandra/pull/2
